### PR TITLE
Added option to use AWS bundled ca cert Fixes #171

### LINKED ISSUE #171: Mac OS X : certificate verify failed (Seahorse::Client::NetworkingError)
Hello,

On Mac OS X I have the following error :
```
/Users/llange/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/net/http.rb:923:in `connect': SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed (Seahorse::Client::NetworkingError)
	from /Users/llange/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/net/http.rb:923:in `block in connect'
	from /Users/llange/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/timeout.rb:89:in `block in timeout'
	from /Users/llange/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/timeout.rb:99:in `call'
	from /Users/llange/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/timeout.rb:99:in `timeout'
	from /Users/llange/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/net/http.rb:923:in `connect'
	from /Users/llange/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/net/http.rb:863:in `do_start'
	from /Users/llange/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/net/http.rb:858:in `start'
...
```

It seems linked with the [following issue of aws-sdk-core](https://github.com/aws/aws-sdk-core-ruby/issues/166#issuecomment-111603660) and the fix suggested does indeed work.

If there is no adverse impact, could you please add the suggested fix ?

I added it to master/lib/terraforming.rb, just after the "require":
```
@@ -2,6 +2,7 @@
 require "ox"

 require "aws-sdk-core"
+Aws.use_bundled_cert!

 require "erb"
 require "json"
```

Thanks.

### PR DESCRIPTION
**Pull request by mattgartman**

### CODE REVIEW

File: lib/terraforming/cli.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ class CLI < Thor
     class_option :tfstate, type: :boolean, desc: "Generate tfstate"
     class_option :profile, type: :string, desc: "AWS credentials profile"
     class_option :region, type: :string, desc: "AWS region"
+    class_option :"aws-use-bundled-cert", type: :boolean , desc: "Use the bundled CA certificate from AWS SDK"
```
Comment (dtan4) [inline lib/terraforming/cli.rb]: __[rubocop]__
* Space found before comma.
* Line is too long. [109/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: lib/terraforming/cli.rb [hunk 2]
```diff
@@ -5,6 +5,9 @@ class CLI < Thor
     class_option :tfstate, type: :boolean, desc: "Generate tfstate"
     class_option :profile, type: :string, desc: "AWS credentials profile"
     class_option :region, type: :string, desc: "AWS region"
+    class_option :"aws-use-bundled-cert",
```
Comment (dtan4) [inline lib/terraforming/cli.rb]: `class_option :use_bundled_cert` looks better.

Underscores are converted to hyphens automatically.

```
  [--use-bundled-cert], [--no-use-bundled-cert]  # Use the bundled CA certificate from AWS SDK
```

**Comment by dtan4** [comment]
Sorry for late review 🙇 IMO `aws-` prefix is unnecessary, the same as `region`, `profile`.

File: lib/terraforming/resource/lambda_function.rb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        http_client.use_ssl = true
+        http_client.ca_file = Aws.config[:ssl_ca_bundle]
+        http_client.verify_mode = OpenSSL::SSL::VERIFY_PEER
+
+        http_client.start do |http|
+          response = http.get(uri)
+
+          unless response.code == "200"
+            raise "Error downloading Lambda Code HTTP Res Code #{response.code} from #{url}"
```
Comment (dtan4) [inline lib/terraforming/resource/lambda_function.rb]: __[rubocop]__
* Line is too long. [92/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: spec/lib/terraforming/resource/lambda_function_spec.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            }
+        ]
+      end
+
+      context "without vpc" do
+        before do
+          puts "================"
+          puts lambdas
+          puts "=================="
+          client.stub_responses(:list_functions, functions:lambdas)
```
Comment (dtan4) [inline spec/lib/terraforming/resource/lambda_function_spec.rb]: __[rubocop]__
* Space missing after colon. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)

File: spec/lib/terraforming/resource/lambda_function_spec.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
```
Comment (dtan4) [inline spec/lib/terraforming/resource/lambda_function_spec.rb]: __[rubocop]__
* Extra empty line detected at block body end.

File: spec/lib/terraforming/resource/lambda_function_spec.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    role = ""
+    handler = ""
+    description = ""
+    memory_size = ""
+    runtime = ""
+    timeout = ""
+    source_code_hash = ""
+}
+          EOF
+
```
Comment (dtan4) [inline spec/lib/terraforming/resource/lambda_function_spec.rb]: __[rubocop]__
* Extra empty line detected at block body end.

**Comment by mattgartman** [comment]
ugh was working on another branch to add lambda support, must have committed to the wrong branch.

**Comment by mattgartman** [comment]
ok, all better.. removed aws- prefix..

File: lib/terraforming/cli.rb [hunk 7]
```diff
@@ -5,6 +5,9 @@ class CLI < Thor
     class_option :tfstate, type: :boolean, desc: "Generate tfstate"
     class_option :profile, type: :string, desc: "AWS credentials profile"
     class_option :region, type: :string, desc: "AWS region"
+    class_option :"use-bundled-cert",
```
Comment (dtan4) [inline lib/terraforming/cli.rb]: Please use `:use_bundled_cert`.

**Comment by dtan4** [comment]
Just one more thing, please update README. There is still `--aws-use-bundled-cert`.

File: README.md [hunk 8]
```diff
@@ -68,7 +68,7 @@ $ terraforming s3 --profile hoge
 You can force the AWS SDK to utilize the CA certificate that is bundled with the SDK for systems where the default OpenSSL certificate is not installed (e.g. Windows) by utilizing the `--aws-use-bundled-cert` option.
```
Comment (dtan4) [inline README.md]: ☝️ one more ☝️

**Comment by dtan4** [comment]
LGTM 👍 Thanks ❗

**Comment by mattgartman** [comment]
Thanks @dtan4!  Sorry for making you triple check this.

**Comment by dtan4** [comment]
No problem 👌 ❗

**Pull request merged**